build-in-container/mariner-docker-builder.sh

```diff
@@ -88,7 +88,7 @@ validate_custom_repo_file() {
     done
 }
 
-tool_dir=$( realpath "$(dirname "$0")" )
+tool_dir=$(realpath $(dirname "${BASH_SOURCE[0]}"))
```

Comment:
Please add an explanation on why this needs to change to using `${BASH_SOURCE[0]}` rather than `$0`. I didn't see any info about it in the PR description.